spec/models/admin_user_spec.rb

```diff
@@ -0,0 +1,40 @@
+require 'rails_helper'
+
+describe AdminUser, type: :model do
+  
+  it 'should be of class AdminUser' do
+    expect(subject.class).to eq AdminUser
+  end
+  
+  it 'should be valid' do
+    expect(FactoryGirl.create(:admin_user)).to be_valid
+  end
+  
+  context 'should not have an invalid email address' do
+    emails = ['asdf@ ds.com', '@example.com', 'test me @yahoo.com', 'asdf@example', 'ddd@.d. .d', 'ddd@.d' ]
+    emails.each do |email|
+      it { is_expected.not_to allow_value(email).for(:email) }
+    end
+  end
+
+  context 'should have a valid email address' do
+    emails = ['asdf@ds.com', 'hello@example.uk', 'test1234@yahoo.si', 'asdf@example.eu' ]
+    emails.each do |email|
+      it { is_expected.to allow_value(email).for(:email) }
+    end
+  end
+  
+  it { is_expected.not_to allow_value("").for(:password) }
+
+  it { should validate_length_of(:password).is_at_least(6) }
```

Comment:
Let's use `is_expected` format instead of `should` format to avoid deprecation warnings.